src/hiprtc.cpp

```diff
@@ -143,7 +143,9 @@ struct _hiprtcProgram {
     {
         using namespace std;
 
-        name = hip_impl::demangle(name.c_str());
+        char* demangled = hip_impl::demangle(name.c_str());
+        name.assign(demangled == nullptr ? std::string() : demangled);
+        std::free(demangled);
```

Comment:
`free` would segfault if `demangled==nullptr`.

Reply:
As opposed the the our previous assumptions, free(nullptr) is a no-op. So a ifcheck is redundant

Reply:
just looked up the behaviour for std::free and yes indeed